streamer/src/socket.rs

```diff
@@ -0,0 +1,25 @@
+use std::net::SocketAddr;
+
+// TODO: remove these once IpAddr::is_global is stable.
+
+#[cfg(debug_assertions)]
+pub fn is_global(_: &SocketAddr) -> bool {
+    true
+}
+
+#[cfg(not(debug_assertions))]
+pub fn is_global(addr: &SocketAddr) -> bool {
+    use std::net::IpAddr;
+
+    !match addr.ip() {
+        IpAddr::V4(addr) => {
+            addr.is_private()
```

Comment:
`is_private` should be ok, no?

Reply:
done

Reply:
I meant that, `is_private()` should be allowed, but the other ranges should not.